server/backends/memory_cache/memory_cache.go

```diff
@@ -31,6 +36,22 @@ func sizeFn(value interface{}) int64 {
 	return size
 }
 
+func Register(env environment.Env) error {
+	if !*cacheInMemory || env.GetCache() != nil {
+		return nil
+	}
```

Comment:
I think we should always warn people when we detect what we'd consider a "misconfiguration" like two caches being configured.

It's sort of preventative medicine in the sense that it will help our customers figure out their own misconfigurations and fix them before they have to reach out to us and ask "how to i setup bb".